users/tests/test_oauth_login.py

```diff
@@ -0,0 +1,251 @@
+from django.test import TestCase
+from django.contrib.auth import get_user_model
+from rest_framework.test import APIClient
+from rest_framework_simplejwt.tokens import RefreshToken
+import requests_mock
+from ..models import UserRole
+from django.urls import reverse
+
+User = get_user_model()
+
+class OAuthTokenObtainPairViewTests(TestCase):
+    """
+    Test suite for the extended OAuthTokenObtainPairView that handles both
+    password and OAuth (Google/GitHub) authentication.
+    """
```

Comment:
Expand the docstring to include what endpoints are tested, and any notable edge cases covered.